pkg/vendir/directory/staging_dir.go

```diff
@@ -103,7 +103,12 @@ type StagingTempArea struct {
 var _ ctlfetch.TempArea = StagingTempArea{}
 
 func (d StagingTempArea) NewTempDir(name string) (string, error) {
-	tmpDir := filepath.Join(d.path, name)
+	dir := d.path
+	if dir == "" {
+		// same behavior as ioutil.TempFile()
+		dir = os.TempDir()
+	}
+	tmpDir := filepath.Join(dir, name)
```

Comment:
we want to use staging dir as temp directory instead of OS specific location because we want to track all files we create in one place. given that there is no need to add this code (especially that we fixed tmpFile.Close() issue).

Reply:
yeah I know, its just that I don't agree. 

Having this check doesn't do harm and if someone forgets to set the path by accident in the future, then maybe it could prevent an other issue like #67.

This kind of defensive programming is used all over in go (and other languages) and as I already said you are using methods in this project that fallback to os.TempDir() already.

Anyway, this kind of stuff is not worth arguing over so I reverted the change.